build.gradle

```diff
@@ -372,13 +375,43 @@ task sourcesJar(type: Jar) {
     classifier = 'sources'
 }
 
-// This is a hack to disable the java 8 default javadoc lint until we fix the html formatting
 if (JavaVersion.current().isJava8Compatible()) {
     tasks.withType(Javadoc) {
-        options.addStringOption('Xdoclint:none', '-quiet')
+        // do this for all javadoc tasks, inlcuding gatkDocs
+        options.addStringOption('Xdoclint:none')
     }
 }
 
+javadoc {
+    // This is a hack to disable the java 8 default javadoc lint until we fix the html formatting
+    // We only want to do this for the javadoc task, not gatkDoc
+    options.addStringOption('Xdoclint:none', '-quiet')
+}
+
+// Generate GATK Online Doc
+task gatkDoc(type: Javadoc, dependsOn: shadowJar) {
+    source = sourceSets.main.allJava
+    classpath = sourceSets.main.runtimeClasspath
+
+    // Since the custom doclet we run is basically another entry point for gatk, it needs all of the gatk
+    // dependencies on the docletpath (i.e., Barclay , etc), so we depend on and use the shadowJar. Also,
+    // add the resources folder to ensure log4j is configured.
+    options.docletpath = [file(shadowJar.archivePath)] + files("src/main/resources")
+    options.doclet = "org.broadinstitute.hellbender.utils.help.GATKHelpDoclet"
+
+    // ensure the target folder already exists
+    File outDir = new File("./gatkDoc")
+    outDir.mkdirs()
```

Comment:
It fires after the configuration phase is complete and the entire task graph is determined, so you can test if a particular task will actually execute:

```
gradle.taskGraph.whenReady { graph ->
    if (graph.hasTask(gatkDoc) && !gatkDocDir.exists() && !gatkDocDir.mkdirs()) {
        throw new GradleException("Can't create output folder for gatkDoc task: " + gatkDocDir.absolutePath)
    }
}
```
I think doFirst makes more sense though, even if it is a little weird that the result is a closure with a doFirst block that executes AFTER the part of the closure that is NOT in the doFirst block!):

```
task gatkDoc {
    doFirst {
        System.out.println("This doesn't happen first")
    }
    System.out.println("This happens first")
}
```